src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -121,6 +131,10 @@ object Task {
     // Regular expression for matching taskIds before instance-era
     private val LegacyTaskIdRegex = """^(.+)[\._]([^_\.]+)$""".r
 
+    // Regular expression for matching resident app taskIds
+    private val ResidentTaskIdRegex = """^(.+)([\._])([^_\.]+)(\.\#)(\d+)$""".r
+    private val ResidentTaskIdAttemptSeparator = ".#"
```

Comment:
I've not used angular, but it looks like `task_id` is part of a "route"...

https://github.com/apache/mesos/blob/8447d196a8d2385d8dd742c5d6a8ac62de9a59fe/src/webui/master/static/js/app.js#L48

.. and stackoverflow says that hashtags are special?

https://stackoverflow.com/questions/27226009/is-angularjs-routes-adding-a-special-character